### ISSUE TITLE
[pre-1.0.8-1] Segmentation fault when processing new tx

### ISSUE THREAD

**Issue by bitkevin** [zcash/zcash#2481]
* Ubuntu 14.04.5 LTS
  * Linux ali.bj.dev.04.btc.com 4.4.0-63-generic #84~14.04.2-Ubuntu SMP Mon Feb 13 11:04:58 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
* Zcash Daemon version v1.0.8-gf630519d86

This segmentation fault is repeatable. Could happen every few days.

Last N lines of debug.log

```
2017-06-24 00:12:44 CreateNewBlock(): total size 25937
2017-06-24 00:13:33 UpdateTip: new best=000000000263b56527aa78b47b85739ce968f7c7edf57c722f4faeb5e11e4c35  height=137183  log2_work=50.163638  tx=771415  date=2017-06-24 00:12:44 progress=0.999997  cache=14.5MiB(830tx)
2017-06-24 00:13:35 CreateNewBlock(): total size 25115
2017-06-24 00:13:45 CreateNewBlock(): total size 25581
2017-06-24 00:13:55 UpdateTip: new best=000000002746647c5719bf56f8ce1516f1936b069704b30e5ca343c19f5eae9d  height=137184  log2_work=50.16367  tx=771418  date=2017-06-24 00:13:33 progress=0.999999  cache=14.5MiB(834tx)
2017-06-24 00:14:00 CreateNewBlock(): total size 23638
2017-06-24 00:14:25 CreateNewBlock(): total size 41835
2017-06-24 00:14:35 CreateNewBlock(): total size 42060
2017-06-24 00:14:45 CreateNewBlock(): total size 45790
2017-06-24 00:15:05 CreateNewBlock(): total size 46016
2017-06-24 00:15:15 CreateNewBlock(): total size 46241
```

GDB stack:

```
[New Thread 0x7ffff46ea7c0 (LWP 27646)]
[New Thread 0x7fff2e5b0700 (LWP 27647)]
[Thread 0x7fff2e5b0700 (LWP 27647) exited]
[Thread 0x7ffff46ea7c0 (LWP 27646) exited]
[New Thread 0x7ffff46ea7c0 (LWP 7389)]
[New Thread 0x7fff2e5b0700 (LWP 7390)]
[Thread 0x7fff2e5b0700 (LWP 7390) exited]
[Thread 0x7ffff46ea7c0 (LWP 7389) exited]

Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fff78acd700 (LWP 26169)]
TxConfirmStats::NewTx (this=this@entry=0x555555e57f10, nBlockHeight=nBlockHeight@entry=137184, val=val@entry=3.7466257302537374e+24) at policy/fees.cpp:251
251         unconfTxs[blockIndex][bucketindex]++;
(gdb) bt
#0  TxConfirmStats::NewTx (this=this@entry=0x555555e57f10, nBlockHeight=nBlockHeight@entry=137184, val=val@entry=3.7466257302537374e+24) at policy/fees.cpp:251
#1  0x00005555557119c9 in CBlockPolicyEstimator::processTransaction (this=0x555555e57db0, entry=..., fCurrentEstimate=fCurrentEstimate@entry=true) at policy/fees.cpp:385
#2  0x00005555556e1663 in CTxMemPool::addUnchecked (this=this@entry=0x555555dfb6a0 <mempool>, hash=..., entry=..., fCurrentEstimate=<optimized out>) at txmempool.cpp:111
#3  0x00005555555fba1e in AcceptToMemoryPool (pool=..., state=..., tx=..., fLimitFree=fLimitFree@entry=true, pfMissingInputs=pfMissingInputs@entry=0x7fff78acba3a,
    fRejectAbsurdFee=fRejectAbsurdFee@entry=false) at main.cpp:1245
#4  0x000055555560233d in ProcessMessage (pfrom=pfrom@entry=0x7fff5c022ca0, strCommand="tx", vRecv=..., nTimeReceived=nTimeReceived@entry=1498263434047340) at main.cpp:4722
#5  0x0000555555605e5f in ProcessMessages (pfrom=0x7fff5c022ca0) at main.cpp:5240
#6  0x0000555555607d6c in boost::detail::function::function_invoker1<bool (*)(CNode*), bool, CNode*>::invoke (function_ptr=..., a0=<optimized out>)
    at /work/zhibiao/zec/zcash-1.0.8/depends/x86_64-unknown-linux-gnu/share/../include/boost/function/function_template.hpp:101
#7  0x000055555567b15e in operator() (a0=<optimized out>, this=<optimized out>)
    at /work/zhibiao/zec/zcash-1.0.8/depends/x86_64-unknown-linux-gnu/share/../include/boost/function/function_template.hpp:771
#8  m_invoke<boost::function<bool(CNode*)>, 0u, CNode*&> (args=..., func=..., this=<optimized out>)
    at /work/zhibiao/zec/zcash-1.0.8/depends/x86_64-unknown-linux-gnu/share/../include/boost/signals2/detail/variadic_slot_invoker.hpp:98
#9  operator()<boost::function<bool(CNode*)>, CNode*&, 1ul> (args=..., func=..., this=<optimized out>)
    at /work/zhibiao/zec/zcash-1.0.8/depends/x86_64-unknown-linux-gnu/share/../include/boost/signals2/detail/variadic_slot_invoker.hpp:90
#10 operator()<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<boost::signals2::detail::slot_meta_group, boost::optional<int> >, boost::signals2::slot<b
ool(CNode*), boost::function<bool(CNode*)> >, boost::signals2::mutex> > > (connectionBody=..., this=<optimized out>)
    at /work/zhibiao/zec/zcash-1.0.8/depends/x86_64-unknown-linux-gnu/share/../include/boost/signals2/detail/variadic_slot_invoker.hpp:134
#11 dereference (this=0x7fff78acca30) at /work/zhibiao/zec/zcash-1.0.8/depends/x86_64-unknown-linux-gnu/share/../include/boost/signals2/detail/slot_call_iterator.hpp:110
#12 dereference<boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<bool, CNode*>, std::_List_iterator<boost::shared_ptr<boost::sign
als2::detail::connection_body<std::pair<boost::signals2::detail::slot_meta_group, boost::optional<int> >, boost::signals2::slot<bool(CNode*), boost::function<bool(CNode*)> >
---Type <return> to continue, or q <return> to quit---
, boost::signals2::mutex> > >, boost::signals2::detail::connection_body<std::pair<boost::signals2::detail::slot_meta_group, boost::optional<int> >, boost::signals2::slot<boo
l(CNode*), boost::function<bool(CNode*)> >, boost::signals2::mutex> > > (f=...)
    at /work/zhibiao/zec/zcash-1.0.8/depends/x86_64-unknown-linux-gnu/share/../include/boost/iterator/iterator_facade.hpp:549
#13 operator* (this=0x7fff78acca30) at /work/zhibiao/zec/zcash-1.0.8/depends/x86_64-unknown-linux-gnu/share/../include/boost/iterator/iterator_facade.hpp:655
#14 operator()<boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<bool, CNode*>, std::_List_iterator<boost::shared_ptr<boost::signa
ls2::detail::connection_body<std::pair<boost::signals2::detail::slot_meta_group, boost::optional<int> >, boost::signals2::slot<bool(CNode*), boost::function<bool(CNode*)> >,
 boost::signals2::mutex> > >, boost::signals2::detail::connection_body<std::pair<boost::signals2::detail::slot_meta_group, boost::optional<int> >, boost::signals2::slot<bool
(CNode*), boost::function<bool(CNode*)> >, boost::signals2::mutex> > > (last=..., first=..., this=<optimized out>) at net.h:95
#15 operator()<CombinerAll, boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<bool, CNode*>, std::_List_iterator<boost::shared_ptr
<boost::signals2::detail::connection_body<std::pair<boost::signals2::detail::slot_meta_group, boost::optional<int> >, boost::signals2::slot<bool(CNode*), boost::function<boo
l(CNode*)> >, boost::signals2::mutex> > >, boost::signals2::detail::connection_body<std::pair<boost::signals2::detail::slot_meta_group, boost::optional<int> >, boost::signal
s2::slot<bool(CNode*), boost::function<bool(CNode*)> >, boost::signals2::mutex> > > (last=..., first=..., combiner=..., this=<optimized out>)
    at /work/zhibiao/zec/zcash-1.0.8/depends/x86_64-unknown-linux-gnu/share/../include/boost/signals2/detail/result_type_wrapper.hpp:53
#16 boost::signals2::detail::signal_impl<bool (CNode*), CombinerAll, int, std::less<int>, boost::function<bool (CNode*)>, boost::function<bool (boost::signals2::connection c
onst&, CNode*)>, boost::signals2::mutex>::operator()(CNode*) (this=<optimized out>, args#0=args#0@entry=0x7fff5c022ca0)
    at /work/zhibiao/zec/zcash-1.0.8/depends/x86_64-unknown-linux-gnu/share/../include/boost/signals2/detail/signal_template.hpp:247
#17 0x0000555555662b24 in operator() (args#0=0x7fff5c022ca0, this=0x555555e4cf78 <g_signals+24>)
    at /work/zhibiao/zec/zcash-1.0.8/depends/x86_64-unknown-linux-gnu/share/../include/boost/signals2/detail/signal_template.hpp:723
#18 ThreadMessageHandler () at net.cpp:1660
#19 0x000055555566ae51 in TraceThread<void (*)()> (name=0x555555a3a6a4 "msghand", func=0x55555566297d <ThreadMessageHandler()>) at util.h:235
#20 0x00005555556664fb in operator()<void (*)(char const*, void (*)()), boost::_bi::list0> (a=<synthetic pointer>, f=<optimized out>, this=<optimized out>)
---Type <return> to continue, or q <return> to quit---
    at /work/zhibiao/zec/zcash-1.0.8/depends/x86_64-unknown-linux-gnu/share/../include/boost/bind/bind.hpp:319
#21 operator() (this=<optimized out>) at /work/zhibiao/zec/zcash-1.0.8/depends/x86_64-unknown-linux-gnu/share/../include/boost/bind/bind.hpp:1294
#22 boost::detail::thread_data<boost::_bi::bind_t<void, void (*)(char const*, void (*)()), boost::_bi::list2<boost::_bi::value<char const*>, boost::_bi::value<void (*)()> >
> >::run (this=<optimized out>) at /work/zhibiao/zec/zcash-1.0.8/depends/x86_64-unknown-linux-gnu/share/../include/boost/thread/detail/thread.hpp:116
#23 0x0000555555855139 in thread_proxy ()
#24 0x00007ffff79bc184 in start_thread (arg=0x7fff78acd700) at pthread_create.c:312
#25 0x00007ffff6ab6bed in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111
```

**Comment by daira** [comment]
This bug was fixed in Zcash v1.0.8-1. Please upgrade to the latest version, v1.0.10-1.

**Comment by daira** [comment]
Was fixed by #2259. @bitkevin, please reopen if you still see this bug after upgrading.
